/**:
  ros__parameters:

    Loc:
      EnableFlag: true
      loadPCDDirectory: "/D:/RepoMy/save/lt/pings/"  # msubmap
      # surroundingKeyframeSearchRadius: 20.0 # 定位时使用的查找范围
      # surroundingKeyframeSearchMaxNum: 10   # 最大取多少个最近帧进行匹配 
      surroundingKeyframeSearchRadius: 20.0 # 定位时使用的查找范围
      surroundingKeyframeSearchMaxNum: 5   # 最大取多少个最近帧进行匹配 
      useSCReLoc: false
      # init_guess: [0.0, 0.0, 3.14, -7.48, -6.655, 0.] 
      init_guess: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0] 
    useRviz: true


    # Topics
    pointCloudTopic: "livox/lidar"                   # Point cloud data
    imuTopic: "livox/imu"                         # IMU data
    odomTopic: "odometry/imu"                    # IMU pre-preintegration odometry, same frequency as IMU
    gpsTopic: "odometry/gpsz"                    # GPS odometry topic from navsat, see module_navsat.launch file

    # Frames
    lidarFrame: "livox_frame"
    baselinkFrame: "base_link"
    odometryFrame: "odom"
    mapFrame: "map"

    # GPS Settings
    useImuHeadingInitialization: false           # if using GPS data, set to "true"
    useGpsElevation: false                       # if GPS elevation is bad, set to "false"
    gpsCovThreshold: 2.0                         # m^2, threshold for using GPS data
    poseCovThreshold: 25.0                       # m^2, threshold for using GPS data

    # Export settings
    savePCD: true                               # https://github.com/TixiaoShan/LIO-SAM/issues/3
    savePCDDirectory: "/D:/RepoMy/save/lt/temp/"         # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation

    # Sensor Settings
    sensor: livox                               # lidar sensor type, either 'velodyne', 'ouster' or 'livox'
    N_SCAN: 4                                   # number of lidar channels (i.e., Velodyne/Ouster: 16, 32, 64, 128, Livox Horizon: 6)
    Horizon_SCAN: 4000                            # lidar horizontal resolution (Velodyne:1800, Ouster:512,1024,2048, Livox Horizon: 4000)
    downsampleRate: 1                           # default: 1. Downsample your data if too many
    # points. i.e., 16 = 64 / 4, 16 = 16 / 1
    lidarMinRange: 2.0                           # default: 1.0, minimum lidar range to be used
    lidarMaxRange: 100.0                        # default: 1000.0, mid-360: 40m@10%  70m@80%  

    # IMU Settings
    imuAccNoise: 3.9939570888238808e-03
    imuGyrNoise: 1.5636343949698187e-03
    imuAccBiasN: 6.4356659353532566e-05
    imuGyrBiasN: 3.5640318696367613e-05

    imuGravity: 9.80511
    imuRPYWeight: 0.01

    extrinsicTrans:  [ -0.011, -0.02329, 0.04412 ]
    extrinsicRot:    [1.0,  0.0,  0.0,
                       0.0,  1.0,  0.0,
                       0.0,  0.0, 1.0 ]
    extrinsicRPY: [1.0, 0.0, 0.0,
                  0.0, 1.0, 0.0,
                  0.0, 0.0, 1.0]

    # LOAM feature threshold
    edgeThreshold: 1.0
    surfThreshold: 0.1
    edgeFeatureMinValidNum: 10
    surfFeatureMinValidNum: 100

    # voxel filter paprams
    odometrySurfLeafSize: 0.2                     # default: 0.4 - outdoor, 0.2 - indoor
    mappingCornerLeafSize: 0.1                    # default: 0.2 - outdoor, 0.1 - indoor
    mappingSurfLeafSize: 0.2                      # default: 0.4 - outdoor, 0.2 - indoor

    # robot motion constraint (in case you are using a 2D robot)
    z_tollerance: 1000.0                          # meters
    rotation_tollerance: 1000.0                   # radians

    # CPU Params
    numberOfCores: 8                              # number of cores for mapping optimization
    mappingProcessInterval: 0.09                  # 0.15 seconds, regulate mapping frequency

    # Surrounding map
    surroundingkeyframeAddingDistThreshold: 0.2   # 2. meters, regulate keyframe adding threshold
    surroundingkeyframeAddingAngleThreshold: 0.05  # 0.2 radians, regulate keyframe adding threshold
    surroundingKeyframeDensity: 2.0               # meters, downsample surrounding keyframe poses   
    surroundingKeyframeSearchRadius: 20.0         # 50. meters, within n meters scan-to-map optimization
    # (when loop closure disabled)

    # Loop closure
    loopClosureEnableFlag: true
    loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency
    surroundingKeyframeSize: 20                   # 50. submap size (when loop closure enabled)
    historyKeyframeSearchRadius: 15.0             # meters, key frame that is within n meters from
    # current pose will be considerd for loop closure
    historyKeyframeSearchTimeDiff: 30.0           # seconds, key frame that is n seconds older will be
    # considered for loop closure
    historyKeyframeSearchNum: 10                  # 25. number of hostory key frames will be fused into a
    # submap for loop closure
    historyKeyframeFitnessScore: 0.3              # icp threshold, the smaller the better alignment

    # Visualization
    globalMapVisualizationSearchRadius: 1000.0    # meters, global map visualization radius
    globalMapVisualizationPoseDensity: 10.0       # meters, global map visualization keyframe density
    globalMapVisualizationLeafSize: 1.0           # meters, global map visualization cloud density
